document/js-api/index.bs

```diff
@@ -1488,11 +1489,10 @@ To <dfn>create a suspending function</dfn> from a JavaScript function |func|, wi
         1. If |ret|.\[[Type]] is <emu-const>throw</emu-const>, then:
             1. Let |type|, |payload| and |opaqueData| be the result of [=coerce a JavaScript exception|coercing the JavaScript exception=] |ret|.\[[Value]].
             1. [=WebAssembly/Throw=] with |type|, |payload| and |opaqueData|.
-        1. Otherwise, if [=list/size=] of |ret| is 1 and [$IsPromise$](|ret|.\[[Value]][0]):
-            1. Let |promise| be |ret|.\[[Value]][0].
+        1. Otherwise,
+            1. Let |promise| be [=?=] [$PromiseResolve$]({{%Promise%}},|ret|.\[[Value]]).
```

Comment:
The `?` macro here is incorrect because it would propagate a throw completion to the caller, which in this case is Wasm. You should use `[$Completion$](PromiseResolve(...))` and do the manual checking, coercion of the exception, and `[=WebAssembly/Throw=]` like done elsewhere in this algorithm.

Reply:
Ack.